File: C++/Arrays/Easy/Largest_element_in_array.cpp
#include <iostream>
#include<bits/stdc++.h>
using namespace std;

int largest_element(vector<int> &arr, int size){
    
    // Brute force

    // 1. Sort Elements in acending/descending Order
    // 2. Take the last/first element

    sort(arr.begin(), arr.end());
    return arr[size - 1];

    // Time Complexity : O(nlogn)
    // Space Complexity : O(1)

}

int largest_element_iterative(vector<int> arr, int size){

    // Iterative Approach

    // 1. Set max = min(integer value) i.e. INT_MIN
    // 2. If max > element of the array update max to that element

    int max = INT_MIN;
    int i = 0;
    while(i < size){
        if (arr[i] > max) max = arr[i];
        i++;
    }
    return max;
}

int main(){

    vector<int> arr = { 5, 0, 11, 2, 3, 24, 1 };
    int n = arr.size();

    int result = largest_element(arr, n);
    cout << "Largest element: " << result << endl;

    int answer = largest_element_iterative(arr, n);
    cout << "Largest element: " << answer << endl;
    return 0;
}

File: Python/Arrays/Easy/Largest_element_in_array.py

def largest_element(arr, size):

    # Brute force

    # 1. Sort Elements in acending/descending Order
    # 2. Take the last/first element
    arr.sort()
    return arr[size - 1]

def largest_element_iterative(arr, size):
    max_num = -float("inf")
    i = 0
    while i < size:
        if arr[i] > max_num : 
            max_num = arr[i]
        i += 1
    return max_num

if __name__ == '__main__':
    arr = [ 5, 0, 11, 2, 3, 24, 1 ]
    n = len(arr)

    result = largest_element(arr, n)
    print('Largest Element in array: ', result)

    answer = largest_element_iterative(arr, n)
    print('Largest Element in array: ', answer)